### ISSUE TITLE
Pull requests should trigger CI builds for all submitters

### ISSUE THREAD

**Issue by jeking3** [StackStorm/st2-docker#150]
Currently it looks like CircleCI builds are only running for pull requests by @warrenvw.  They should run for all submitters.

**Comment by LindsayHill** [comment]
CircleCI settings changed

It was not restricted to @warrenvw. The setting is related to builds from forked PRs.

**Comment by jeking3** [comment]
The branch history in the repo is getting pretty dirty.  I'd recommend everyone start using personal forks and submitting PRs from them, and clean out all the development branches.

**Comment by LindsayHill** [comment]
Are you referring to the 7 stale branches that should have been deleted?
What impact does a ‘dirty branch history’ have?

**Comment by jeking3** [comment]
It looks like folks started developing this project on the same fork, using branches to stage changes.  It looks like folks with write access continue to use branches on the main project instead of in their own fork.  This has a number of consequences:

1. This means when people fetch, they get development branches in varying states (unnecessarily).  
2. Settings on the project may work for branch builds within the repo but not for PR builds coming in from a fork of the repo.  (This happened here...)
3. Settings for CI on forks of the repo might not work at all if assumptions were made about the branches being in the same repo.  (I've seen this happen in boost project repositories)
4. "git branch -a" shows more information than just release branches.

I have found that it's best if folks with commit access to the repo follow the same procedures as folks without commit access for submitting changes, as it forces the committers to make the public code submission and project management (for example I may want to enable CI on my fork) work for everyone.

The st2 main repository has the same issue.  So again, it's no big deal, just a best practice I've adopted in other projects and wanted to share.
